server/mscalendar/oauth2.go

```diff
@@ -72,6 +95,18 @@ func (app *oauth2App) CompleteOAuth2(authedUserID, code, state string) error {
 		return err
 	}
 
+	uid, err := app.Store.LoadMattermostUserId(me.ID)
+	if err == nil {
```

Comment:
When `err == nil`, that means that we asserted that the user is already connected, which is not a good thing, since we are trying to (re)connect the user. So this block is something that should reject the user's intent to reconnect to the same account. As a result, we tell them to disconnect first.